index.d.ts

```diff
@@ -1,3 +1,31 @@
+export interface TimeoutOptions {
+	/**
+	Milliseconds before timing out.
+
+	Passing `Infinity` will cause it to never time out.
+
+	@default Infinity
+	*/
+	milliseconds?: number;
+
+	/**
+	Specify a custom error message or error.
+
+	If you do a custom error, it's recommended to sub-class `TimeoutError`.
+	*/
+	message?: string | Error;
+
+	/**
+	Custom implementations for the `setTimeout` and `clearTimeout` functions.
+
+	Useful for testing purposes, in particular to work around [`sinon.useFakeTimers()`](https://sinonjs.org/releases/latest/fake-timers/).
+	*/
+	customTimers?: {
+		setTimeout: typeof globalThis.setTimeout;
+		clearTimeout: typeof globalThis.clearTimeout;
```

Comment:
This suggests that if you specify the object, both are mandatory. You'll likely have to add a question mark on both lines

Reply:
@fregante 
Maybe it's good to ensure that `customTimers`'s type is the same as `p-timeout`'s `customTimers`'s type.
How about like the below one?

```ts
import { Options as pTimeoutOptions } from 'p-timeout';

export interface TimeoutOptions {
	...
	customTimers?: pTimeoutOptions['customTimers']
}
```